packages/build-utils/src/prerender.ts

```diff
@@ -165,5 +168,27 @@ export class Prerender {
       }
       this.experimentalStreamingLambdaPath = experimentalStreamingLambdaPath;
     }
+
+    if (chain !== undefined) {
+      if (typeof chain !== 'object') {
+        throw new Error(
+          'The `chain` argument for `Prerender` must be an object.'
+        );
+      }
+
+      if (!chain.headers || typeof chain.headers !== 'object') {
```

Comment:
It might be good to validate the headers entries are correct shape as well similar to `initialHeaders` validation